File: BinaryTrees/FindLevel/target.py
class Solution:

    def getTargetLevel(self, node, target, level):
        if not node:
            return -1

        if node.data == target:
            return level

        left = self.getTargetLevel(node.left, target, level + 1)
        if left != -1:
            return left

        return self.getTargetLevel(node.right, target, level + 1)

    def getLevel(self, node, target):
        level = self.getTargetLevel(node, target, 1)
        return level

File: BinaryTrees/FindLevel/target.cpp
#include <iostream>
using namespace std;

class Node {
public:
    int data;
    Node* left;
    Node* right;
    Node(int val) {
        data = val;
        left = nullptr;
        right = nullptr;
    }
};

class Solution {
  public:
    int getTargetLevel(struct Node *node, int target,int level){
        if(!node)return -1;
        if(node->data == target)return level;
        return max(getTargetLevel(node->left,target,level+1),getTargetLevel(node->right,target,level+1));
    }
    int getLevel(struct Node *node, int target) {
        // code here
        return 1+getTargetLevel(node,target,0);
    }
};